### ISSUE TITLE
[Bug]: DocumentDB cluster parameter group apply_method not taking effect immediately

### ISSUE THREAD

**Issue by skurtEnd** [hashicorp/terraform-provider-aws#28352]
### Terraform Core Version

v1.3.5

### AWS Provider Version

4.43.0

### Affected Resource(s)

aws_docdb_cluster_parameter_group

### Expected Behavior

Should create DocumentDB ttl_monitoring enabled with apply_method taking effect immediately

Expecting to see 'No infrastructure changes' when running terraform plan

### Actual Behavior

Terraform plan always shows changes after apply:

```
+ parameter {
          + apply_method = "immediate"
          + name         = "ttl_monitor"
          + value        = "enabled"
        }
      - parameter {
          - apply_method = "pending-reboot" -> null
          - name         = "ttl_monitor" -> null
          - value        = "enabled" -> null
        }
```

### Terraform Configuration Files

```
resource "aws_docdb_cluster_parameter_group" "this" {
  family                 = "family"
  name                  = "name"
  description        = "description"

  dynamic "parameter" {
    content {
      apply_method   = "immediate"
      name                  = "ttl_monitor"
      value                   = "enabled"
    }
  }
```

### Steps to Reproduce

terraform init
terraform apply
terraform plan

### Important Factoids

The only way to not see that diff in the plan is by switching the value from enabled to disabled and then back to enabled again.

So you apply with the following values:

```
dynamic "parameter" {
    content {
      apply_method   = "immediate"
      name                  = "ttl_monitor"
      value                   = "disabled"
    }
  }
```

Then

```
dynamic "parameter" {
    content {
      apply_method   = "immediate"
      name                  = "ttl_monitor"
      value                   = "enabled"
    }
  }
```

Then it works fine, however it doesn't enable ttl_monitor from the get-go

### Would you like to implement a fix?

None

**Comment by justinretzolk** [comment]
Potentially related: #22857

**Comment by justinretzolk** [comment]
It looks like this is a behavior of the underlying API, and not something that can be corrected in the Terraform AWS Provider (see #24737, which added a note to the documentation around this limitation, for more information). With that in mind, since there's no further action that we can take, we'll close this issue. If the API changes at some point, and allows us to correct this, we'd be happy to take another look.
